sdk/core/core/inc/az_span.h

```diff
@@ -304,6 +304,29 @@ AZ_NODISCARD az_result az_span_atou64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_atou32(az_span span, uint32_t* out_number);
 
+/**
+ * @brief Parses an #az_span containing ASCII digits into positive or negative double number. This
+ * function uses atof from standard library, so it will convert digits until the last valid
+ * character is found.
+ *
+ * If there is not an invalid character before the end of \p content, function will use
+ * \p conversion_buffer to copy \p span content and attach a \0 at the end to set
+ * the end of the digits.
+ *
+ * Use NULL for \p conversion_buffer if \p span content has already an invalid character or \0
+ * after the last digit to convert.
+ *
+ * @param[in] span The #az_span containing the ASCII digits to be parsed.
+ * @param[in] conversion_buffer \p span content will be copied here and a \0 is appended at the end
```

Comment:
This is a somewhat complicated name. What `conversion`? How about `copy_buffer`?